vscode-lean4/src/infoview.ts

```diff
@@ -568,23 +578,29 @@ export class InfoProvider implements Disposable {
         if (kind === 'above') {
             // in this case, assume that we actually want to insert at the same
             // indentation level as the neighboring text
-            const prev_line = editor.document.lineAt(pos.line - 1);
-            const spaces = prev_line.firstNonWhitespaceCharacterIndex;
+            const lines = text.split('\n');
+            const last_line = lines.length - 1;
+            const length_of_last_line = lines[last_line].length;
+            const current_line = editor.document.lineAt(pos.line);
+            const spaces = current_line.firstNonWhitespaceCharacterIndex;
             const margin_str = [...Array(spaces).keys()].map(x => ' ').join('');
-
             let new_command = text.replace(/\n/g, '\n' + margin_str);
-            new_command = `\n${margin_str}${new_command}`;
+            new_command = `${margin_str}${new_command}\n`;
+            const insertPosition = current_line.range.start;
 
             await editor.edit((builder) => {
-                builder.insert(prev_line.range.end, new_command);
+                builder.insert(insertPosition, new_command);
             });
-            editor.selection = new Selection(pos.line, spaces, pos.line, spaces);
+            // select the whole insert so it is easy for user to delete it.
```

Comment:
This is quite annoying (at least with the extensions that I use) because I first need to clear the selection before I can type something again.  Immediately deleting the inserted text is certainly not the intended use case.

Reply:
You'd typically have to move the cursor for your next edit anyway, and that would clear the selection.

Reply:
Not with the vim extension at least.

Reply:
I also think it is good manners when inserting new text to make it easy to delete it if the user decides that's not what they wanted although I suppose undo would also do that.

Reply:
The selection also makes it more visible what we just did int the copy to comment command.

Reply:
and it makes it easy to move the pasted comment :-) all that for a tiny inconvenience of having to hit right arrow... :-)

Reply:
After having just tried it, I am *personally* slightly in favour of not adding the selection as the window switch with `preserveFocus: false` seems quite sufficient to indicate what the extension did. But! The "personally" is important ─ it seems hard to decide this kind of thing without doing a user study which we certainly don't want to do for such a tiny feature. So ultimately I would be okay with either option.

Reply:
I'd also be happy to to get more user feedback on this before deciding the issue.

Reply:
Anecdotally, I don't think anybody has ever complained about the current behavior either.  I'd really prefer to keep the behavior change out of this bug fix.

> all that for a tiny inconvenience of having to hit right arrow... :-)

It's escape with the vim extension.  And you don't expect to land in visual mode since pretty much no other text insertion (such as paste) changes the selection.

And you found the right word: it's an inconvenience.  The change is a trade-off.  At best, we're making one workflow easier and another one more complicated.  At worst, users need to relearn how to use the extension just for change's sake.

Reply:
Ok, I removed the selection change.  In fact, I removed all selection movement to so your cursor stays exactly where it was.